.github/workflows/ci.yml

```diff
@@ -153,45 +167,44 @@ jobs:
       #   run: BRANCH=${{ github.ref_name }} PULL_REQUEST=${{ github.event.number }} scripts/ci/upload_logs_to_s3 ./yarn-project/end-to-end/log
 
   # barretenberg (prover) native tests
+  # only ran on x86 for resource reasons (memory intensive)
   bb-native-tests:
     needs: setup-x86
-    runs-on: ${{ github.actor }}-${{matrix.environment}}
+    runs-on: ${{ github.actor }}-x86
+    timeout-minutes: 15
     strategy:
       fail-fast: false
-      matrix:
-        environment: [x86]
-        # pending fix for intermittent test
-        # environment: [x86, arm]
     steps:
       - {uses: actions/checkout@v4, with: { ref: "${{ github.event.pull_request.head.sha }}"}}
       - {uses: ./.github/ci-setup-action, with: { dockerhub_password: "${{ secrets.DOCKERHUB_PASSWORD }}"}}
-      # # Only allow one memory-hunger prover test to use this runner
-      # # As detailed in https://github.com/ben-z/gh-action-mutex
-      # # things do not become 'pending' in github actions, and instead just cancel one another
-      # # so we can't use the native concurrency in GA
-      # - name: Set up mutex
-      #   uses: ben-z/gh-action-mutex@v1.0.0-alpha.9
-      #   with:
-      #     branch: gh-action-mutex-bench-${{ github.actor }}
+      # Only allow one memory-hunger prover test to use this runner
+      # As detailed in https://github.com/ben-z/gh-action-mutex
+      # things do not become 'pending' in github actions, and instead just cancel one another
```

Comment:
Unfortunately have to use a somewhat janky custom action that locks with a branch. works, though